llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -660,6 +743,14 @@ static bool expandIntrinsic(Function &F, CallInst *Orig) {
   case Intrinsic::dx_radians:
     Result = expandRadiansIntrinsic(Orig);
     break;
+  case Intrinsic::dx_resource_load_typedbuffer:
+    expandTypedBufferLoadIntrinsic(Orig);
+    Orig->eraseFromParent();
+    return true;
+  case Intrinsic::dx_resource_store_typedbuffer:
+    expandTypedBufferStoreIntrinsic(Orig);
+    Orig->eraseFromParent();
+    return true;
```

Comment:
This feels like a bit of a break in abstraction from the other intrinsics here and I think it would be surprising that these just return if we were to add any other logic after the switch here. Can we adjust the structure of this function so that we don't need to do a special case here?

Reply:
Yes, I think so.